core/controllers/classroom_test.py

```diff
@@ -381,6 +517,118 @@ def test_update_classroom_data(self) -> None:
 
         self.logout()
 
+    def test_update_classroom_thumbnail_data_only(
+        self
+    ) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        classroom_handler_url = '%s/%s' % (
+            feconf.CLASSROOM_HANDLER_URL, self.physics_classroom_id)
+        csrf_token = self.get_new_csrf_token()
+
+        self.physics_classroom_dict['thumbnail_data']['filename'] = 'update.svg'
+
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'test_svg.svg'),
+            'rb', encoding=None
+        ) as f:
+            raw_thumbnail_image = f.read()
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            'rb', encoding=None
+        ) as f:
+            raw_banner_image = f.read()
+
+        params = {'payload': json.dumps({
+            'classroom_dict': self.physics_classroom_dict
+        })}
+        params['csrf_token'] = csrf_token
+
+        thumbnail = (
+            'thumbnail_image', 'thumbnail_filename1', raw_thumbnail_image)
+        banner = ('banner_image', 'banner_filename1', raw_banner_image)
+        self.testapp.put(
+            classroom_handler_url,
+            params=params, expect_errors=False,
+            upload_files=[thumbnail, banner]
+        )
+
+        # Check physics classroom data updated correctly.
+        physics_classroom = classroom_config_services.get_classroom_by_id(
+            self.physics_classroom_id
+        )
+        self.assertEqual(
+            physics_classroom.to_dict(), self.physics_classroom_dict)
+
+        # Check new thumbnail image uploaded correctly.
+        fs = fs_services.GcsFileSystem(
+            feconf.ENTITY_TYPE_CLASSROOM, self.physics_classroom_id)
+        self.assertTrue(fs.isfile('thumbnail/%s' % 'update.svg'))
```

Comment:
Here and elsewhere, you don't need the %s. Just inline the string directly into the string template. Only use the % when you are trying to include a variable.

Reply:
Got it. I've updated this test and the one below.